buildkite/client.go

```diff
@@ -28,3 +31,32 @@ func NewClient(org, apiToken string) *Client {
 		organization: org,
 	}
 }
+
+func (client *Client) makeRequest(method string, url string, postData interface{}, responseObject interface{}) error {
+	jsonPayload, err := json.Marshal(postData)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest(method, url, bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.http.Do(req)
+	if err != nil && resp.StatusCode >= 400 {
+		return err
+	}
```

Comment:
my spacing here is stuffed, but I think if we check for non-200 status code independently of the err, then my scope failure will be fatal.

```suggestion
  if err != nil {
    return err
  }
  if resp.StatusCode >= 400 {
    return fmt.Errorf("buildkite API request failed: %s %s (status: %d)", method, url, resp.StatusCode)
  }
```